src/poliastro/ground_station.py

```diff
@@ -0,0 +1,145 @@
+import numpy as np
+from astropy import time, units as u
+
+
+class GroundStation(object):
+    """ Class representing a ground station on an arbitrary ellipsoid.
+    """
+
+    def __init__(self, lon, lat, h, epoch, angular_velocity, a, b, c=None):
+        """
+
+        Parameters
+        ----------
+        lon: ~astropy.units.quantity.Quantity
+            geodetic longitude
+
+        lat: ~astropy.units.quantity.Quantity
+            geodetic latitude
+
+        h: ~astropy.units.quantity.Quantity
+            geodetic height
+
+        epoch: ~astropy.units.quantity.Quantity
+            starting epoch
+
+        angular_velocity: ~astropy.units.quantity.Quantity
+            angular velocity
+
+        a: ~astropy.units.quantity.Quantity
+            semi-major axis
+
+        b: ~astropy.units.quantity.Quantity
+            semi-minor axis
+
+        c: ~astropy.units.quantity.Quantity
+            Third axis defining the ellipsoid.
+            If not set, it defaults to the value of a, making the ellipsoid
+            a spheroid.
+        """
+        self._lon = lon
+        self._lat = lat
+        self._h = h
+        self._epoch = epoch
+        self._angular_velocity = angular_velocity
+        self._a = a
+        self._b = b
+        self._c = c if c is not None else a
+
+    @property
+    def cartesian_cords(self):
+        """Convert to the Cartesian Coordinate system.
+        """
+        e2 = 1 - (self._b / self._a) ** 2
+        N = self._a / np.sqrt(1 - e2 * np.sin(self._lon) ** 2)
+
+        x = (N + self._h) * np.cos(self._lon) * np.cos(self._lat)
+        y = (N + self._h) * np.cos(self._lon) * np.sin(self._lat)
+        z = ((1 - e2) * N + self._h) * np.sin(self._lon)
+        return x, y, z
+
+    @property
+    def f(self):
+        """Get first flattening.
+        """
+        return 1 - self._b / self._a
+
+    @property
+    def N(self):
+        """Normal vector of the ellipsoid at the point of the ground station.
+        """
+        x, y, z = self.cartesian_cords
+        a, b, c = self._a.value, self._b.value, self._c.value
+        N = np.array([2 * x.value / a ** 2, 2 * y.value / b ** 2, 2 * z.value / c ** 2])
+        N /= np.linalg.norm(N)
+        return N
+
+    @property
+    def tangential_vecs(self):
+        """Returns orthonormal vectors tangential to the ellipsoid at the point of the ground station.
+        """
+        N = self.N
+        u = np.array([1.0, 0, 0])
+        u -= u.dot(N) * N
+        u /= np.linalg.norm(u)
+        v = np.cross(N, u)
+        return u, v
+
+    def propagate(self, t):
```

Comment:
I am not exacly sure what `gs.propagate` is supposed to do, but it doesn't seem quite right to me. What do you think will be the use case?

Reply:
To clarify: to me, a `GroundStation` is something static, attached to the surface of something. Surely you can propagate that something, but propagating the ground station still sounds weird.

Reply:
I agree and I actually spend a lot of time thinking about it, but couldn't come up with anything better. Now we may want to account for the rotation of the body, so there should be a way to specify said rotation in a given epoch or after a certain interval. That being said, calling ``gs.rotate(time)`` sounds even less reasonable.

But, there may be a way to deal with this if we allow for a ``Body`` as you mentioned bellow, since we could then use the body itself and not the ground station (this could probably require some work on the ``Body`` class itself). Alternatively, we could use something simple like ``specify_epoch(time)`` which makes more sense.